rust/lance/src/dataset/scanner.rs

```diff
@@ -1584,9 +1585,10 @@ impl Scanner {
         projection: &Schema,
         batch_readahead: usize,
     ) -> Result<Arc<dyn ExecutionPlan>> {
+        let coalesced = Arc::new(CoalesceBatchesExec::new(input, self.get_batch_size()));
```

Comment:
Do we want to sort row IDs to offer a better chance we can do sequential reads?

Reply:
We already do that internally.